src/dds/build/plan/library.cpp

```diff
@@ -71,6 +89,12 @@ library_plan library_plan::create(const library_root&             lib,
         compile_rules.include_dirs().push_back(codegen_subdir);
     }
 
+    auto public_header_compile_rules          = compile_rules.clone();
+    public_header_compile_rules.syntax_only() = true;
+    auto src_header_compile_rules             = public_header_compile_rules.clone();
+    lib.append_private_compile_rules(compile_rules);
+    lib.append_private_compile_rules(src_header_compile_rules);
```

Comment:
These lines are both `append_private_compile_rules`. Typo?

Reply:
I see that the function naming is poor. It's appending to both `compile_rules` and `src_header_compile_rules`, since both of those need `-I src`.

Reply:
Ahh, that clears it up.

In a future iteration, the way compile rules are passed and generated will be cleaned up significantly, so such confusion won't be necessary.

Reply:
Try looking at `neo::into` and `neo::output<T>`, which can be used to make output parameters apparent and explicit.